File: php/phpsearch/searchoption.php
<?php

class SearchOption {
    function __construct($shortarg, $longarg, $desc, $func) {
        $this->shortarg = $shortarg;
        $this->longarg = $longarg;
        $this->desc = $desc;
        $this->func = $func;
        $this->sortarg = $this->__sortarg();
    }

    private function __sortarg() {
        if ($this->shortarg) {
            return strtolower($this->shortarg) . 'a' . $this->longarg;
        }
        return $this->longarg;
    }
}

?>

File: objc/objcsearch/Sources/objcsearch/SearchOption.m
#import "SearchOption.h"

@implementation SearchOption

- (instancetype) initWithShortArg:(NSString *)sArg withLongArg:(NSString *)lArg withDesc:(NSString *)desc withArgType:(ArgTokenType)argType {
    self = [super init];
    if (self) {
        self.shortArg = sArg;
        self.longArg = lArg;
        self.desc = desc;
        self.argType = argType;
    }
    return self;
}

- (NSString *) sortArg {
    if (self.shortArg) {
        return [NSString stringWithFormat:@"%@a%@", [self.shortArg lowercaseString], [self.longArg lowercaseString]];
    }
    return [self.longArg lowercaseString];
}

- (NSString *) description {
    NSMutableString *d = [[NSMutableString alloc] initWithString:@"SearchOption("];
    [d appendFormat:@"short: \"%@\", long: \"%@\", desc: \"%@\")",
     self.shortArg, self.longArg, self.description];
    return d;
}

@end
